src/tabnote.ts

```diff
@@ -266,13 +278,13 @@ export class TabNote extends StemmableNote {
   }
 
   // Get the fret positions for the note
-  getPositions() {
+  getPositions(): { str: string; fret: string }[] {
     return this.positions;
   }
 
   // Add self to the provided modifier context `mc`
-  addToModifierContext(mc) {
-    this.setModifierContext(mc);
+  addToModifierContext(mc: ModifierContext): this {
+    this.modifierContext = mc;
```

Comment:
Why not `this.setModifierContext`?

Reply:
To let the compiler realize that `this.modifierContext` is not `undefined` avoiding the usage of `!`. 
It is a `protected` member and we save a function call which simply does the assignment.

Reply:
Okay, that's a good reason. Looking at the codebase, it seems like `setModifierContext` doesn't do anything special, and it's only called in three places.

Can you remove `setModifierContext` entirely from `note.ts` and `modifier.ts`, and the references in `stavenote.ts`, `tabnote.ts`, and `modifiercontext.ts` too in this PR please? It's a good cleanup.

Reply:
I have modified `StaveNote` to use the protected member. We cannot remove `setModifierContext` it is used by `ModifierContext` in `addMember`

Reply:
Okay, yes, thanks.